### PR TITLE
[RAG-75] Docs: add langchain evaluation notebook

### PR DESCRIPTION
**Pull request by jordanrfrazier**
Adds a basic langchain evaluation notebook.

### CODE REVIEW

File: examples/notebooks/langchain_evaluation.ipynb [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+   "source": [
+    "# WARNING: This will delete the collection and all documents in the collection\n",
+    "# vstore.delete_collection()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nbmake": {
+     "post_cell_execute": [
```
Comment (epinzur) [inline examples/notebooks/langchain_evaluation.ipynb:677]: not sure if a "post_cell_execute" will run on a "markdown" cell

Comment (jordanrfrazier) [reply examples/notebooks/langchain_evaluation.ipynb:677]: Good catch. Tested this and you are correct.

File: examples/notebooks/langchain_evaluation.ipynb [hunk 2]
```diff
@@ -0,0 +1,719 @@
...
+                "    prediction_key=\"result\",\n",
+                ")\n",
+                "\n",
+                "client = Client()\n",
+                "run_on_dataset(\n",
+                "    dataset_name=dataset_name,\n",
+                "    llm_or_chain_factory=create_qa_chain(llm=model, vstore=vstore),\n",
+                "    client=client,\n",
+                "    evaluation=evaluation_config,\n",
+                "    verbose=True,\n",
```
Comment (epinzur) [inline examples/notebooks/langchain_evaluation.ipynb:647]: maybe re-run with this set `False`, to significantly reduce the output in the notebook?

File: examples/notebooks/langchain_evaluation.ipynb [hunk 3]
```diff
@@ -0,0 +1,719 @@
...
+                "from langsmith import Client\n",
+                "from langchain.evaluation import EvaluatorType\n",
+                "from langchain.smith import RunEvalConfig, run_on_dataset\n",
+                "\n",
+                "evaluation_config = RunEvalConfig(\n",
+                "    # LangChain offers several QA Evaluator types\n",
+                "    evaluators=[\n",
+                "        \"qa\",\n",
+                "        \"context_qa\",\n",
+                "        \"cot_qa\",\n",
```
[the remark this answers is not in the document]
Comment (epinzur) [reply examples/notebooks/langchain_evaluation.ipynb:636]: Alternatively maybe we should show using Ragas for the evaluators, and not the langchain ones... the langchain ones are really limited.

like the setup from this blog: https://blog.langchain.dev/evaluating-rag-pipelines-with-ragas-langsmith/

**Review by epinzur — APPROVED** [review]

**Pull request merged**